sig-observability/concepts-and-designs/notification-manager-v3.1.0.md

```diff
@@ -0,0 +1,517 @@
+- WebhookReceiver: Define the WebhookConfig selector.
+
+ The relationship between receivers and configs can be demonstrated as below:
+
+ ![Receivers & Configs](https://raw.githubusercontent.com/kubesphere/notification-manager/master/docs/images/receivers_configs.png)
+
+ Receiver CRDs like EmailReceiver, WechatReceiver, SlackReceiver and WebhookReceiver can be categorized into 2 types `global` and `tenant` by label like `type = global`, `type = tenant` :
+- A global EmailReceiver receives all alerts and then sends notifications regardless tenant info(user or namespace).
+- A tenant EmailReceiver receives alerts with specified tenant labels like `user` or `namespace`
+
+ Usually alerts received from Alertmanager contains a `namespace` label, Notification Manager uses this label to decide which receiver to use for sending notifications:
+- For KubeSphere, Notification Manager will try to find workspace `user` in that `namespace`'s rolebinding and then find receivers with `user = xxx` label.
+- For other Kubernetes cluster, Notification Manager will try to find receivers with `namespace = xxx` label. 
+
+ For alerts without a `namespace` label, for example, alerts of node or kubelet, user can setup a receiver with `type = global` label to receive alerts without a `namespace` label. A global receiver sends notifications for all alerts received regardless any label. A global receiver is usually set for an admin role.
+
+ Config CRDs like EmailConfig, WechatConfig, SlackConfig, WebhookConfig can be categorized into 2 types `tenant` and `default` by label like `type = tenant`, `type = default`:
+- Tenant EmailConfig is to be selected by a tenant EmailReceiver which means each tenant can have his own EmailConfig. 
+- If no EmailConfig selector is configured in an EmailReceiver, then this EmailReceiver will try to find a `default` EmailConfig. Usually, the admin will set a global default config.
+
+ A receiver could be configured without xxxConfigSelector, in which case Notification Manager will try to find a default xxxConfigSelector with `type = default` label, for example:
+- A global EmailReceiver with `type = global` label should always use the default EmailConfig which means emailConfigSelector needn't be configured for a global EmailReceiver and one default EmailConfig with `type = default` label needs to be configured for all global EmailReceivers.  
+- Usually a tenant EmailReceiver with `type = tenant` label could have its own tenant emailConfigSelector to find its tenant EmailConfig with `type = tenant` label.
+- A tenant EmailReceiver with `type = tenant` label can also be configured without an emailConfigSelector, in which case Notification Manager will try to find the default EmailConfig with `type = default` label for this tenant EmailReceiver.
+
+
+## Demand ##
+
+- User can define the channels for receiving notifications, including email¡¢wechat¡¢slack¡¢webhook¡¢dingtalk,  by creating the corresponding receiver and config.
```

Comment:
email¡¢wechat¡¢slack¡¢webhook¡¢dingtalk =>
email, wechat, slack, webhook and dingtalk